apps/setting/models_provider/base_model_provider.py

```diff
@@ -105,7 +109,7 @@ def filter_optional_params(model_kwargs):
 class BaseModelCredential(ABC):
 
     @abstractmethod
-    def is_valid(self, model_type: str, model_name, model: Dict[str, object], provider, raise_exception=True):
+    def is_valid(self, model_type: str, model_name, model: Dict[str, object], model_params, provider, raise_exception=True):
         pass
 
     @abstractmethod
```

Comment:
Here are some points to consider for your code:

1. The `get_model_credential` method returns the model's credential. You might want to add more descriptive comments or documentation.

2. The parameter name `provider` is not used anywhere within the method. Consider using a meaningful argument name that conveys its purpose.

3. In both methods `is_valid_credential`, you're passing the same `self` argument. This could be redundant and can improve performance slightly if removed.

4. The `get_model_params` method currently retrieves the entire model information, which includes the credentials. It would be better to use this method to retrieve just the parameters rather than accessing them directly from the instance variable.

5. If the model parameters differ based on various factors (e.g., context), it might make sense to encapsulate these dependencies in a separate class. However, without more details about how they change, this isn't strictly necessary here.

Overall, your code is mostly clean with minor improvements possible related to naming conventions and readability.